hazelcast/src/main/java/com/hazelcast/config/CredentialsFactoryConfig.java

```diff
@@ -115,4 +108,21 @@ public int hashCode() {
         result = 31 * result + (properties != null ? properties.hashCode() : 0);
         return result;
     }
+
+    @Override
+    public ICredentialsFactory asCredentialsFactory(ClassLoader cl) {
```

Comment:
Can this method been called from multiple threads concurrently?

Reply:
It probably can. :+1:
I'll make the fields volatile.